### DISCUSSION TITLE
Global OS property

### DISCUSSION THREAD

**Discussion by NotSpaulding** [JanDeDobbeleer/oh-my-posh#2325] [Ideas]
In oh-my-posh you can add custom prompt symbols with an OS segment, for example:
```json
        {
          "foreground": "default",
          "style": "plain",
          "type": "os",
          "properties": {
            "windows": ">",
            "linux": "{{ if .Root }}#{{ else }}${{ end }}",
            "macos": "%"
          },
          "template": "{{ .Icon }} "
        }
```

Sadly you can only do something like this in a segment, which is not supported in transient prompts or even secondary prompts! I think oh-my-posh would really benefit a global OS property that can provide you basic information about your OS.

So maybe if the OS is MacOS it returns a 0, if linux returns a 1... I don't see how else this could be implemented in oh-my-posh, considering you oh-my-posh doesn't really allow for if statements with strings, which is what I am after with this.

**Comment by JanDeDobbeleer** [comment]
Undocumented in this case, so I'll add that, but you can use the following template:

```json
"template": "{{ if eq .OS \"darwin\" }}%{{ else if eq .OS \"windows\" }}>{{ else }}{{ if .Root }}#{{ else }}${{ end }}{{ end }}"
```

**Reply by NotSpaulding** [reply]
Ah I see. How come it's not documented already though?

**Reply by JanDeDobbeleer** [reply]
Because I forgot? I'm only human. It's merged and will be available in the template [documentation](https://ohmyposh.dev/docs/configuration/templates#global-properties) in a few minutes.

**Reply by NotSpaulding** [reply]
Ah alright lol. No problem with that. I thought you just decided not to for some reason. Thanks for your help!
